services/fission-user-workers/service/ctrl_docker.go

```diff
@@ -0,0 +1,252 @@
+package userworkers
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/client"
+	"github.com/google/uuid"
+	"github.com/samber/lo"
+
+	"sensorbucket.nl/sensorbucket/internal/env"
+	"sensorbucket.nl/sensorbucket/internal/pagination"
+)
+
+type DockerController struct {
+	docker client.APIClient
+	store  Store
+
+	workersEndpoint string
+	workerImage     string
+	workerNetID     string
+	amqpHost        string
+	amqpExchange    string
+	endpointDevices string
+}
+
+func CreateDockerController(store Store) (*DockerController, error) {
+	docker, err := client.NewClientWithOpts(client.FromEnv)
+	if err != nil {
+		return nil, err
+	}
+	// Find network id
+	netID := env.Could("CTRL_DOCK_WORKER_NET", "")
+	if netID == "" {
+		// try to get it ourselves
+		nets, err := docker.NetworkList(context.Background(), types.NetworkListOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("not network id given and an error occured while trying to get a network: %w", err)
+		}
+		for _, net := range nets {
+			if strings.Contains(strings.ToLower(net.Name), "sensorbucket") {
+				netID = net.ID
+				break
+			}
+		}
+	}
+	ctrl := &DockerController{
+		docker: docker,
+		store:  store,
+
+		workersEndpoint: env.Could("CTLR_DOCK_WORKERS_EP", "http://caddy/api/workers"),
+		workerImage:     env.Could("CTLR_DOCK_WORKER_IMAGE", "sensorbucket/docker-worker:latest"),
+		workerNetID:     netID,
+		amqpHost:        env.Could("CTRL_DOCK_AMQP_HOST", "amqp://guest:guest@mq:5672"),
+		amqpExchange:    env.Could("CTRL_DOCK_AMQP_XCHG", "pipeline.messages"),
+		endpointDevices: env.Could("CTRL_DOCK_ENDPOINT_DEVICES", "http://caddy/api/devices"),
+	}
+	return ctrl, nil
+}
+
+func (ctrl *DockerController) Reconcile(ctx context.Context) error {
+	log.Println("Starting docker reconciliation...")
+	defer log.Println("Reconciliation finished")
+	filter := filters.NewArgs(
+		filters.Arg("label", "controlled-by=sensorbucket"),
```

Comment:
is there any dynamic thing going in the runtime for the controlled-by label? If so, maybe it should be an env var

Reply:
Nothing at the moment. It is just used to identify kubernetes resources that sensorbucket should control.